source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -137,9 +137,59 @@ void MapdataCheckPoint::clearSearched() {
     m_searched = false;
 }
 
+/// @addr{0x80511EC8}
+/// @brief Finds the offset between the two positions that enter the checkpoint.
+/// @details This assumes the player is entering the checkpoint as intended, and not from the side.
+/// @param prevPos The previous position, likely not located in the checkpoint.
+/// @param pos The current position, likely located in the checkpoint.
+/// @return The earliest subdivision that crosses into the checkpoint, in the range [1, 17].
+u16 MapdataCheckPoint::getEntryOffsetMs(const EGG::Vector2f &prevPos,
+        const EGG::Vector2f &pos) const {
+    constexpr f32 REFRESH_PERIOD = 1000.0f / 59.94f;
+
+    EGG::Vector2f velocity = pos - prevPos;
+    velocity *= 1.0f / REFRESH_PERIOD;
+
+    // d_k = p_0 - m + kv
```

Comment:
What is this formula? Can you elaborate a tiny bit in this comment?

Reply:
There's not a whole lot more I can do to explain it in text beyond what the code describes.